server/pbenchinacan/redeploy

```diff
@@ -0,0 +1,107 @@
+
+# Locations on the host
+GITTOP=${GITTOP:-$(git rev-parse --show-toplevel)}
+PB_DASHBOARD_DIR=${PB_DASHBOARD_DIR:-${GITTOP}/dashboard/build}
+SRV_PBENCH=${SRV_PBENCH:-/srv/pbench}
+
+# Locations inside the container
+#
+# The value of ${NGINX_FAVICON} points to the location where the file will be
+# found inside the container and is used to create a symbolic link; by default,
+# it refers to a file in the Dashboard deployment relative to
+# /srv/pbench/public_html.
+NGINX_FAVICON=${NGINX_FAVICON:-./dashboard/$(realpath --relative-to "${PB_DASHBOARD_DIR}" "${PB_DASHBOARD_DIR}"/static/media/color-square.256x256*.ico)}
+
+#+
+# Deployment
+#-
+
+
+# Stop the currently running Pbench Server, if any.
+systemctl --user stop pbench-server || true
+
+# This is the first invocation of the Podman run command in this script, so we
+# execute a trivial containerized command to make Podman apply the requested
+# pull policy here.  After this invocation (if it doesn't fail...), the
+# container image will be present in the local repository, and so we will use
+# the default Podman policy ("missing") for the subsequent invocations.
+# (Otherwise, we risk pulling a remote image when we shouldn't.)  However, if
+# this script was invoked by the run-pbench-in-a-can script, then _it_ will have
+# done the first pull, and it will have set the policy to "missing" to prompt us
+# to skip it here.  In any case, if the pull policy is set to "missing", then
+# the first Podman run invocation below will do the right thing on its own, so
+# we skip the extra one here.
+if [[ ${PB_SERVER_IMAGE_PULL_POLICY} != "missing" ]]; then
+    podman run --entrypoint true --rm --pull "${PB_SERVER_IMAGE_PULL_POLICY}" "${PB_SERVER_IMAGE}"
+fi
+
+# Update the Dashboard code, removing any existing code and copying in fresh.
+# We expect that the host dashboard directory already exists and contains a
+# build of the Pbench Dashboard application; however, if it is missing, we
+# deploy the Pbench Server without a Dashboard deployment.  If we do update the
+# Dashboard, then also update the Nginx favicon symlink.
+podman unshare rm -r "${SRV_PBENCH}/public_html/dashboard"
+if [[ -d "${PB_DASHBOARD_DIR}" ]]; then
+    podman run \
+        --rm \
+        --volume "${PB_DASHBOARD_DIR}":/dashboard:Z \
+        --volume "${SRV_PBENCH}":/srv/pbench:Z \
+        --entrypoint cp \
+        "${PB_SERVER_IMAGE}" \
+        -r /dashboard /srv/pbench/public_html/dashboard
+    podman run \
+        --rm \
+        --volume "${SRV_PBENCH}":/srv/pbench:Z \
+        --entrypoint ln \
+        "${PB_SERVER_IMAGE}" \
+        -sf "${NGINX_FAVICON}" /srv/pbench/public_html/favicon.ico
+else
+    echo "Warning:  deploying the Pbench Server without a Dashboard" >&2
+    podman run \
+        --rm \
+        --volume "${SRV_PBENCH}":/srv/pbench:Z \
+        --entrypoint mkdir \
+        "${PB_SERVER_IMAGE}" \
+        -m 0755 -p /srv/pbench/public_html/dashboard
+fi
+
+# Run the Pbench Server.
```

Comment:
"Start the Pbench Server service" would be a better description if you happen to walk this trail again ...

Reply:
That was actually my first instinct as well...and then I decided that that particular text didn't add anything to the `systemctl` command which followed it.  So, I concluded that I should either remove the comment or set it to something like what appears above.

Reply:
I disagree, but in that case you might as well delete the comments because they certainly don't add any value.